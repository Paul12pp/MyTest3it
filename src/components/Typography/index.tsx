import React from 'react';
import {Pressable, Text} from 'react-native';
import styles from './style';
import {Props} from './types';

export const Typography = ({
  children = '',
  style = {},
  variant = {type: 'default'},
  bold = false,
  numberOfLines = 0,
  onPress,
  testID = 'typography-test',
}: Props) => {
  const typographyStyle = {
    ...styles[variant.type],
    ...(bold && styles.bold),
    ...style,
  };
  return onPress ? (
    <Pressable onPress={onPress}>
      <Text
        testID={testID}
        style={typographyStyle}
        numberOfLines={numberOfLines}>
        {children}
      </Text>
    </Pressable>
  ) : (
    <Text testID={testID} style={typographyStyle} numberOfLines={numberOfLines}>
      {children}
    </Text>
  );
};
