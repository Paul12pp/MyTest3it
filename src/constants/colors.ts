export const colors = {
  primary: '#2d1e72',
  secondary: '#ab5dec',
  green: '#00ff99',
  blue: '#337ab7',
  blueLight: '#00bcd4',
  blueDark: '#2e6da4',
  carbon: '#253D52',
  white: '#ffffff',
  black: '#000000',
  danger: '#a94442',
  jet: '#333333',
  orange: '#ffd65e',
  golden: '#d7a307',
  neutral: {
    feel: '#050709',
    neutral0: '#FFFFFF', //'#FFFFFF',
    neutral50: '#F6F6F6', //'#FFFFFF',
    neutral100: '#E9EBEC', //TO DO Delete color
    neutral200: '#E2E2E2',
    neutral400: '#ADADAD', //'#A1A2A4',
    neutral600: '#545454', //'#585A5B',
    neutral900: '#141414', //'#050709',
    neutralc4: '#C4C4C4',
  },
};
