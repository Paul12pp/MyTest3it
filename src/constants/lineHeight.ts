export const lineHeight = {
  h1: 64,
  h2: 58,
  h3: 44,
  h4: 34,
  h5: 26,
  h6: 24,
  subtitle: {
    subtitle1: 24,
    subtitle2: 22,
  },
  body: {
    body1: 24,
    body2: 22,
    p2: 20,
  },
  button: 22,
  caption: 16,
  subcaption: 14,
  overline: 16,
  linkTextBig: 24,
  linkTextMedium: 22,
  linkTextSmall: 16,
  bodyLinkMedium: 20,
};
