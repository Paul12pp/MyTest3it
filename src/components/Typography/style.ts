import {StyleSheet} from 'react-native';
import {fontSizes} from '../../constants/fontSizes';
import {letterSpacing} from '../../constants/letterSpacing';
import {lineHeight} from '../../constants/lineHeight';

export default StyleSheet.create({
  default: {
    fontSize: fontSizes.default,
  },
  small: {
    fontSize: fontSizes.small,
  },
  vlarge: {
    fontSize: fontSizes.vlarge,
  },
  vextralarge: {
    fontSize: fontSizes.vextralarge,
  },
  vmextralarge: {
    fontSize: fontSizes.vmextralarge,
  },
  extralarge: {
    fontSize: fontSizes.extralarge,
  },
  large: {
    fontSize: fontSizes.large,
  },
  regular: {
    fontSize: fontSizes.regular,
  },
  medium: {
    fontSize: fontSizes.medium,
  },
  vsmall: {
    fontSize: fontSizes.vsmall,
  },
  vmsmall: {
    fontSize: fontSizes.vmsmall,
  },
  vmmsmall: {
    fontSize: fontSizes.vmmsmall,
  },
  vvsmall: {
    fontSize: fontSizes.vvsmall,
  },
  bold: {
    fontWeight: '700',
  },
  h6: {
    fontSize: fontSizes.regular,
    fontWeight: '700',
    lineHeight: lineHeight.h6,
    letterSpacing: letterSpacing.h6,
  },
  h5: {
    fontSize: fontSizes.default,
    fontWeight: '700',
    lineHeight: lineHeight.h5,
    letterSpacing: letterSpacing.h5,
  },
  h4: {
    fontSize: fontSizes.medium,
    fontWeight: '700',
    lineHeight: lineHeight.h4,
    letterSpacing: letterSpacing.h4,
  },
  body1: {
    fontSize: fontSizes.small,
    lineHeight: lineHeight.body.body1,
    letterSpacing: letterSpacing.body.body1,
  },
  body2: {
    fontSize: fontSizes.vsmall,
    lineHeight: lineHeight.body.body2,
    letterSpacing: letterSpacing.body.body2,
  },
  bodyP1: {
    fontSize: fontSizes.small,
    lineHeight: lineHeight.linkTextBig,
    letterSpacing: letterSpacing.linkTextSmall,
  },
  bodyP2: {
    fontSize: fontSizes.vsmall,
    lineHeight: lineHeight.body.p2,
    letterSpacing: letterSpacing.linkTextSmall,
  },
  subtitleS1: {
    lineHeight: lineHeight.subtitle.subtitle1,
    fontSize: fontSizes.small,
    fontWeight: '700',
    letterSpacing: letterSpacing.linkTextSmall,
  },
  subtitleS2: {
    lineHeight: lineHeight.subtitle.subtitle2,
    fontSize: fontSizes.vsmall,
    fontWeight: '700',
    letterSpacing: letterSpacing.subtitle.subtitle2,
  },
  bodyLinkMedium: {
    fontSize: fontSizes.vsmall,
    lineHeight: lineHeight.bodyLinkMedium,
    letterSpacing: letterSpacing.bodyLinkMedium,
  },
  overline: {
    letterSpacing: letterSpacing.overline,
    fontSize: fontSizes.vmmsmall,
    fontWeight: '700',
  },
  caption: {
    fontSize: fontSizes.vmsmall,
    lineHeight: lineHeight.caption,
    letterSpacing: letterSpacing.caption,
  },
  subcaption: {
    fontSize: fontSizes.vmmsmall,
    lineHeight: lineHeight.caption,
    letterSpacing: letterSpacing.caption,
  },
  subpcaption: {
    fontSize: fontSizes.vvmmsmall,
    lineHeight: lineHeight.subcaption,
    letterSpacing: letterSpacing.subcaption,
  },
});
