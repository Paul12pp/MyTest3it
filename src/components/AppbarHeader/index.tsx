import React from 'react';
import {styles} from './style';
import {AppbarHeaderProps} from './type';
import {Appbar} from 'react-native-paper';
import Icon from 'react-native-vector-icons/MaterialIcons';
import * as RootNavigation from '../../utils/navigationref/RootNavigation';
import {colors} from '../../constants/colors';

const AppbarHeader = ({title, type = 'default'}: AppbarHeaderProps) => {
  const _handleBack = () => {
    if (type === 'back') {
      RootNavigation.goBack();
    }
  };
  return (
    <Appbar.Header>
      <Icon
        style={styles.iconBack}
        color={colors.primary}
        size={24}
        testID={'icon-id'}
        name={type === 'back' ? 'arrow-back' : 'menu'}
        onPress={_handleBack}
      />
      <Appbar.Content
        testID={'content-id'}
        titleStyle={styles.title}
        title={title}
      />
    </Appbar.Header>
  );
};

export default AppbarHeader;
