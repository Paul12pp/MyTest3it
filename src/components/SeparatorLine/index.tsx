import React from 'react';
import {View} from 'react-native';
import {colors} from '../../constants/colors';
import {styles} from './style';
import {Props} from './type';

export const SeparatorLine = ({
  color = colors.neutral.neutral100,
  testID,
  customStyle = {},
}: Props) => {
  return (
    <View
      style={{
        ...styles.line,
        borderColor: color,
        ...customStyle,
      }}
      testID={testID}
    />
  );
};
