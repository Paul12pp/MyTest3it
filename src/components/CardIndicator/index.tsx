import React from 'react';
import {TouchableOpacity, View} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {colors} from '../../constants/colors';
import {SeparatorLine} from '../SeparatorLine';
import {Typography} from '../Typography';
import {styles} from './style';
import {CardIndicatorProps} from './type';

const CardIndicator = ({item, onPress, onPressInfo}: CardIndicatorProps) => {
  // console.log(item);
  return (
    <TouchableOpacity
      testID={'card-test'}
      style={styles.container}
      onPress={onPress}>
      <View style={styles.subContainer}>
        <View>
          <Typography style={styles.title} variant={{type: 'body1'}}>
            {item[1].nombre}
          </Typography>
          <Typography style={styles.subtitle} variant={{type: 'body2'}}>
            {item[1].unidad_medida}
          </Typography>
        </View>
        <View style={styles.boxIcons}>
          <Icon
            name={'information-circle-outline'}
            size={24}
            testID={'info-icon'}
            color={colors.primary}
            onPress={onPressInfo}
          />
          <Icon
            name={'chevron-forward-outline'}
            size={24}
            testID={'arrow-icon'}
            color={colors.neutral.neutral400}
          />
        </View>
      </View>
      <SeparatorLine testID={'separator-line'} />
    </TouchableOpacity>
  );
};

export default CardIndicator;
