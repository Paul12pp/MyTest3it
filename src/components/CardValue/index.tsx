import React from 'react';
import {View} from 'react-native';
import {SeparatorLine} from '../SeparatorLine';
import {Typography} from '../Typography';
import {styles} from './style';
import {CardValueProps} from './type';
import moment from 'moment';

const CardValue = ({item}: CardValueProps) => {
  return (
    <View style={styles.container}>
      <View style={styles.subContainer}>
        <Typography style={styles.date} variant={{type: 'body1'}}>
          {moment(item.fecha).format('YYYY-MM-DD')}
        </Typography>
        <Typography style={styles.value} variant={{type: 'body2'}}>
          $ {item.valor}
        </Typography>
      </View>
      <SeparatorLine />
    </View>
  );
};

export default CardValue;
