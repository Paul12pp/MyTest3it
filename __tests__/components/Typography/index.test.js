import 'react-native';
import React from 'react';
import {Typography} from '../../../src/components/Typography';
import renderer from 'react-test-renderer';
import styles from '../../../src/components/Typography/style';
import {fontFamily} from '../../../src/constants/fontSizes';
import {lineHeight} from '../../../src/constants/lineHeight';
import {letterSpacing} from '../../../src/constants/letterSpacing';

describe('The Typography component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Typography />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with children', () => {
    const tree = renderer
      .create(<Typography>Hello World!</Typography>)
      .toJSON();
    expect(tree.children[0]).toBe('Hello World!');
    expect(tree).toMatchSnapshot();
  });

  it('should change fontSize when variant is small', () => {
    const tree = renderer
      .create(<Typography variant={{type: 'small'}} />)
      .toJSON();
    expect(tree.props.style.fontSize).toBe(styles.small.fontSize);
  });

  it('should change fontSize when variant is large', () => {
    const tree = renderer
      .create(<Typography variant={{type: 'large'}} />)
      .toJSON();
    expect(tree.props.style.fontSize).toBe(styles.large.fontSize);
  });

  it('should change the type when variant is body1', () => {
    const tree = renderer
      .create(<Typography variant={{type: 'body1'}} />)
      .toJSON();
    expect(tree.props.style.lineHeight).toBe(lineHeight.body.body1);
    expect(tree.props.style.letterSpacing).toBe(letterSpacing.body.body1);
  });

  it('should change the type when variant is body2', () => {
    const tree = renderer
      .create(<Typography variant={{type: 'body2'}} />)
      .toJSON();
    expect(tree.props.style.lineHeight).toBe(lineHeight.body.body2);
    expect(tree.props.style.letterSpacing).toBe(letterSpacing.body.body2);
  });

  it('should change the type when variant is h5', () => {
    const tree = renderer
      .create(<Typography variant={{type: 'h5'}} />)
      .toJSON();
    expect(tree.props.style.lineHeight).toBe(lineHeight.h5);
    expect(tree.props.style.letterSpacing).toBe(letterSpacing.h5);
  });

  it('should change the type when variant is h4', () => {
    const tree = renderer
      .create(<Typography variant={{type: 'h4'}} />)
      .toJSON();
    expect(tree.props.style.lineHeight).toBe(lineHeight.h4);
    expect(tree.props.style.letterSpacing).toBe(letterSpacing.h4);
  });

  it('should change the type when variant is overline', () => {
    const tree = renderer
      .create(<Typography variant={{type: 'overline'}} />)
      .toJSON();
    expect(tree.props.style.letterSpacing).toBe(letterSpacing.overline);
  });

  it('should accept custom styles', () => {
    const tree = renderer
      .create(<Typography style={{color: 'red'}} />)
      .toJSON();
    expect(tree.props.style.color).toBe('red');
  });
});
