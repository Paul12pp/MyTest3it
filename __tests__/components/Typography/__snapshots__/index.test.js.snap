// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Typography component should render correctly 1`] = `
<Text
  numberOfLines={0}
  style={
    Object {
      "fontSize": 20,
    }
  }
  testID="typography-test"
/>
`;

exports[`The Typography component should render correctly with children 1`] = `
<Text
  numberOfLines={0}
  style={
    Object {
      "fontSize": 20,
    }
  }
  testID="typography-test"
>
  Hello World!
</Text>
`;
